synthesizer/benches/utilities/workload.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+#![allow(unused)]
+
+use crate::utilities::{
+    construct_next_block,
+    initialize_object_store,
+    initialize_vm,
+    split,
+    Benchmark,
+    BenchmarkTransactions,
+    ObjectStore,
+    Operation,
+};
+
+use console::{
+    account::PrivateKey,
+    network::Testnet3,
+    prelude::Network,
+    program::{Identifier, Literal, Plaintext, Record, Value},
+};
+use snarkvm_synthesizer::{
+    helpers::memory::ConsensusMemory,
+    Block,
+    ConfirmedTransaction,
+    ConsensusStorage,
+    Program,
+    Transaction,
+    VM,
+};
+use snarkvm_utilities::{FromBytes, TestRng, ToBytes};
+
+use anyhow::Result;
+use console::{prelude::IoResult, program::Entry};
+use itertools::Itertools;
+use rand::Rng;
+use std::{
+    borrow::{Borrow, BorrowMut},
+    collections::hash_map::DefaultHasher,
+    hash::Hash,
+    iter,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+/// A batch of benchmarks for the workload.
+pub type BenchmarkBatch = Vec<(String, Vec<Transaction<Testnet3>>)>;
+
+/// A `Workload` is a collection of benchmarks to be run together.
+pub struct Workload {
+    /// The name of the workload.
+    name: String,
+    /// The benchmarks to be run.
+    benchmarks: Vec<Box<dyn Benchmark<Testnet3>>>,
+    /// An object store to cache objects for the workload.
+    object_store: ObjectStore,
+}
+
+impl Workload {
+    /// Constructs a new workload.
+    /// Note that the name of the workload is used to construct a directory to store workload objects.
+    /// The directory is located at `.resources/<name>`.
+    pub fn new(name: String, benchmarks: Vec<Box<dyn Benchmark<Testnet3>>>) -> Result<Self> {
+        // Construct the path to a directory to store workload objects.
+        let mut root = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        root.push(".resources");
+        root.push(&name);
+        // Construct the object store.
+        let object_store = ObjectStore::new(root)?;
+        // Construct the workload.
+        Ok(Self { name, benchmarks, object_store })
+    }
+
+    /// Adds a benchmark to the workload.
+    pub fn add(&mut self, benchmark: Box<dyn Benchmark<Testnet3>>) {
+        self.benchmarks.push(benchmark);
```

Comment:
```suggestion
    pub fn add<T: Benchmark<Testnet3>>(&mut self, benchmark: T) {
        self.benchmarks.push(Box::new(benchmark) as Box<dyn Benchmark<Testnet3>>);
```
I think this should work; it would remove plenty of boilerplate at the callsites